### DISCUSSION TITLE
Navigating from outside a Tab Bar Navigator

### DISCUSSION THREAD

**Discussion by AnesBkdm** [react-navigation/react-navigation#11467] [Q&A]
I would like to maintain full control over the bottom tab bar of my app (to later integrate animations, etc.), so I decided not to define the tab bar in <Tab.Navigator> and create a completely separate View that would act as my tab bar navigator and hopefully control the different displayed screens like we would control states.

I would like to gain access to the navigator prop that is inside the <Tab.Navigator> component and pass it to an external custom Navbar. Is there any way? I've seen that there's a useNavigation hook, but it is only usable in a child component.

Thank you very much.

https://github.com/AnesBkdm/snippets/blob/19f58bc6f7d944cd6843715c58c772c8f545e78b/snippet.js

**Comment by satya164** [comment]
I would suggest you to create a custom navigator https://reactnavigation.org/docs/custom-navigators/
